### ISSUE TITLE
Bind() should guess deserialization method without reading body

### ISSUE THREAD

**Issue by mholt** [martini-contrib/binding#4]
**tl;dr** If a request comes in without a Content-Type, what do we do?

When the Bind() method is [looking at the request to see which deserialization method to use](https://github.com/martini-contrib/binding/blob/master/binding.go#L36-47), there is, I think, a bug if the "else" block is entered... suppose JSON was POSTed without a Content-Type. That "else" block is entered and, first, a JSON deserialization is attempted. If there are _any_ errors, it uses form deserialization.

Two problems:
1. The JSON deserializer reads the body, so the stream is empty if and when the form deserializer tries to read it
2. The JSON deserializer may succeed, even if the Validation middleware later adds errors to the context (for example: "cannot be a negative number" for one of the fields...), so form deserialization proceeds when it should not, because there is at least one error on the context.

I'm willing to submit a fix for these, but would like some feedback first. How do we guess the Content-Type correctly and elegantly? Peek at the first character of the request body for a `[` or `{` character to detect JSON? That seems kind of unreliable.

Thoughts?
